diff --git a/lang/c/src/CMakeLists.txt b/lang/c/src/CMakeLists.txt
index c1761c8de..488e558f5 100644
--- a/lang/c/src/CMakeLists.txt
+++ b/lang/c/src/CMakeLists.txt
@@ -130,6 +130,8 @@ configure_file(avro-c.pc.in avro-c.pc)
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/avro-c.pc
         DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
 
+if(NOT CMAKE_SYSTEM_NAME STREQUAL "iOS")
+
 add_executable(avrocat avrocat.c)
 target_link_libraries(avrocat avro-static)
 install(TARGETS avrocat RUNTIME DESTINATION bin)
@@ -148,3 +150,5 @@ add_executable(avromod avromod.c)
 target_link_libraries(avromod avro-static)
 install(TARGETS avromod RUNTIME DESTINATION bin)
 endif(NOT WIN32)
+
+endif()
